web/src/components/common/MerchantProfile.tsx

```diff
@@ -40,20 +42,27 @@ const DetailsContainer = styled.div`
  * This is the profile of the signed-up merchant, shown on the side bar of merchant pages.
  */
 const MerchantProfile: React.FC = () => {
-  const dummyMerchant = {
-    name: 'Merchant Name',
-    email: 'merchant@email.com',
-    vpa: 'merchantVpa@bank',
-  };
-  // TODO: Get merchant details from context instead of using dummy data.
+  const [merchant, setMerchant] = useState<MerchantResponse | undefined>(
+    undefined
+  );
+  const {getMerchant} = useMerchantContext();
+
+  useEffect(() => {
+    const populateMerchantData = async () => {
+      const merchant = await getMerchant();
+      setMerchant(merchant);
+    };
+
+    populateMerchantData();
+  }, [getMerchant]);
 
   return (
     <CentralisedContainer>
       <Shop />
-      <Header>{dummyMerchant.name}</Header>
+      <Header>{merchant?.name}</Header>
```

Comment:
this prints null if merchant not yet fetched?
how about we move merchant null check to higher level, and only render content after merchant is fetched?

Reply:
@ifhuang okay!